worker/logger/logger_test.go

```diff
@@ -78,21 +79,19 @@ func (s *LoggerSuite) TestRunStop(c *gc.C) {
 }
 
 func (s *LoggerSuite) TestInitialState(c *gc.C) {
-	config, err := s.State.ModelConfig()
-	c.Assert(err, jc.ErrorIsNil)
-	expected := config.LoggingConfig()
+	expected := s.loggerAPI.config
 
 	initial := "<root>=DEBUG;wibble=ERROR"
 	c.Assert(expected, gc.Not(gc.Equals), initial)
 
 	loggo.DefaultContext().ResetLoggerLevels()
-	err = loggo.ConfigureLoggers(initial)
+	err := loggo.ConfigureLoggers(initial)
 	c.Assert(err, jc.ErrorIsNil)
 
 	loggingWorker := s.makeLogger(c)
-	defer worker.Stop(loggingWorker)
-
 	s.waitLoggingInfo(c, expected)
+	worker.Stop(loggingWorker)
```

Comment:
That was the race. The worker may update the value in another goroutine. Stop the worker before confirming the expected callback.